### PR TITLE
add more fields to config including patch

### PR DESCRIPTION
**Pull request by OmgImAlexis**
@sharkykh @p0psicles 

From Slack:
> OmgImAlexis [10:30]
> I want to remove some of the fields from the API such as `PLEX_SERVER_PASSWORD` and other API keys, etc. I'm thinking instead we return a boolean depending on if it's set or not?

### CODE REVIEW

File: medusa/server/api/v2/config.py [hunk 1]
```diff
@@ -53,13 +53,45 @@ class ConfigHandler(BaseRequestHandler):
...
         'torrents.path': BooleanField(app, 'TORRENT_PATH'),
+        'torrents.paused': BooleanField(app, 'TORRENT_PAUSED'),
+        'torrents.rpcurl': StringField(app, 'TORRENT_RPCURL'),
+        'torrents.seedLocation': StringField(app, 'TORRENT_SEED_LOCATION'),
+        'torrents.seedTime': StringField(app, 'TORRENT_SEED_TIME'),
+        'torrents.username': StringField(app, 'TORRENT_USERNAME'),
+        'torrents.verifySSL': BooleanField(app, 'TORRENT_VERIFY_CERT'),
+        'nzb.enabled': BooleanField(app, 'USE_NZBS'),
+        'nzb.username': StringField(app, 'NZBGET_USERNAME'),
+        'nzb.password': StringField(app, 'NZBGET_PASSWORD'),
```
Comment (sharkykh) [inline medusa/server/api/v2/config.py]: These two should be removed (duplicates of `'nzb.nzbget.username'` and `'nzb.nzbget.password'`)

File: medusa/server/api/v2/config.py [hunk 2]
```diff
@@ -196,53 +228,77 @@ def data_main():
...
         section_data['torrents']['rpcurl'] = app.TORRENT_RPCURL
-        section_data['torrents']['authType'] = app.TORRENT_AUTH_TYPE
+        section_data['torrents']['seedLocation'] = app.TORRENT_SEED_LOCATION
+        section_data['torrents']['seedTime'] = app.TORRENT_SEED_TIME
+        section_data['torrents']['username'] = app.TORRENT_USERNAME
+        section_data['torrents']['verifySSL'] = bool(app.TORRENT_VERIFY_CERT)
+
         section_data['nzb'] = NonEmptyDict()
         section_data['nzb']['enabled'] = bool(app.USE_NZBS)
         section_data['nzb']['username'] = app.NZBGET_USERNAME
```
Comment (sharkykh) [inline medusa/server/api/v2/config.py]: Same, duplicate of: `section_data['nzb']['nzbget']['username']`

File: medusa/server/api/v2/config.py [hunk 3]
```diff
@@ -196,53 +228,77 @@ def data_main():
...
-        section_data['nzb']['password'] = app.NZBGET_PASSWORD
-        # app.NZBGET_CATEGORY
-        # app.NZBGET_CATEGORY_BACKLOG
-        # app.NZBGET_CATEGORY_ANIME
-        # app.NZBGET_CATEGORY_ANIME_BACKLOG
-        section_data['nzb']['host'] = app.NZBGET_HOST
-        section_data['nzb']['priority'] = app.NZBGET_PRIORITY
+        # section_data['nzb']['password'] = app.NZBGET_PASSWORD
```
Comment (sharkykh) [inline medusa/server/api/v2/config.py]: Duplicate of `section_data['nzb']['nzbget']['password']`

File: themes-default/slim/static/js/store.js [hunk 4]
```diff
@@ -82,14 +82,15 @@ const store = new Puex({
...
+                dir: null,
+                enabled: null,
+                highBandwidth: null,
                 label: null,
-                paused: null,
                 method: null,
+                paused: null,
+                rpcurl: null,
+                seedTime: null,
                 verifySSL: null
```
Comment (sharkykh) [inline themes-default/slim/static/js/store.js]: seems to be missing the `username` property

File: themes-default/slim/static/js/store.js [hunk 5]
```diff
@@ -104,10 +105,26 @@ const store = new Puex({
...
                 password: null,
-                enabled: null
+                sabnzbd: {
+                    category: null,
+                    forced: null,
+                    categoryAnime: null,
+                    categoryBacklog: null,
+                    categoryAnimeBacklog: null
+                },
+                username: null
```
Comment (sharkykh) [inline themes-default/slim/static/js/store.js]: No need for this property

File: themes-default/slim/static/js/store.js [hunk 6]
```diff
@@ -104,10 +105,26 @@ const store = new Puex({
...
+                nzbget: {
+                    category: null,
+                    categoryAnime: null,
+                    categoryAnimeBacklog: null,
+                    categoryBacklog: null,
+                    priority: null,
+                    useHttps: null,
+                    username: null
+                },
                 password: null,
```
Comment (sharkykh) [inline themes-default/slim/static/js/store.js]: No need for this property.

File: themes-default/slim/static/js/store.js [hunk 7]
```diff
@@ -104,10 +105,26 @@ const store = new Puex({
...
-                enabled: null
+                sabnzbd: {
+                    category: null,
+                    forced: null,
+                    categoryAnime: null,
+                    categoryBacklog: null,
+                    categoryAnimeBacklog: null
```
Comment (sharkykh) [inline themes-default/slim/static/js/store.js]: Seems to be missing:
```
username
host
```

**Comment by OmgImAlexis** [comment]
@sharkykh did I miss something? The tests aren't passing and I don't really know enough about the Python ones to work out the issue.

File: dredd/api-description.yml [hunk 8]
```diff
@@ -1278,8 +1278,8 @@ definitions:
                 type: string
               host:
                 type: string
-              password:
-                type: string
+              # password:
```
Comment (p0psicles) [inline dredd/api-description.yml]: Do you need to keep these commented?

File: medusa/server/api/v2/config.py [hunk 9]
```diff
@@ -196,53 +226,94 @@ def data_main():
...
+        section_data['kodi']['notify']['snatch'] = bool(app.KODI_NOTIFY_ONSNATCH)
+        section_data['kodi']['notify']['download'] = bool(app.KODI_NOTIFY_ONDOWNLOAD)
+        section_data['kodi']['notify']['subtitleDownload'] = bool(app.KODI_NOTIFY_ONSUBTITLEDOWNLOAD)
+        section_data['kodi']['update'] = NonEmptyDict()
+        section_data['kodi']['update']['library'] = bool(app.KODI_UPDATE_LIBRARY)
+        section_data['kodi']['update']['full'] = bool(app.KODI_UPDATE_FULL)
+        section_data['kodi']['update']['onlyFirst'] = bool(app.KODI_UPDATE_ONLYFIRST)
+        section_data['kodi']['host'] = app.KODI_HOST
+        section_data['kodi']['username'] = app.KODI_USERNAME
+        # section_data['kodi']['password'] = app.KODI_PASSWORD
```
Comment (p0psicles) [inline medusa/server/api/v2/config.py]: Imo we can leave these out.
Can you remove it, in stead of commenting?

Comment (p0psicles) [reply medusa/server/api/v2/config.py]: I don't mind leaving out the passwords. But then please remove the commented lines.

Comment (OmgImAlexis) [reply medusa/server/api/v2/config.py]: I left the comments I could find out if we needed them removed or not.

**Pull request merged**